diff --git a/constraint_analyzer/traverse_db_schema.rb b/constraint_analyzer/traverse_db_schema.rb
index f29062d..65b63d3 100644
--- a/constraint_analyzer/traverse_db_schema.rb
+++ b/constraint_analyzer/traverse_db_schema.rb
@@ -235,11 +235,20 @@ def traverse_all_for_db_schema(app_dir, interval = nil)
   # newest versions come first
   versions.each_cons(2).each do |newv, curv|
     this_version_has = Hash.new 0
+    shortv = shorten_commit(newv.commit)
+    shortvo = shorten_commit(curv.commit)
     newv.compare_db_schema(curv) do |action, table, *args|
       case action
       when :col_ren, :col_del, :col_type, :col_add
         col = args[0]
         column_changes[table][col] += 1 unless action == :col_add
+        puts "#{shortvo} #{shortv} \e[31;1m#{action}\e[37;0m #{table} #{args}" if action == :col_del
+      when :fk_del, :has_one_del, :has_many_del
+        # puts "#{shortvo} #{shortv} \e[31;1m#{action}\e[37;0m #{table} #{args[0]}"
+      when :assoc_change
+        # puts "#{shortvo} #{shortv} \e[33;1m#{action}\e[37;0m #{table} #{args[0]} #{args[-1]} → #{args[-2]}"
+      when :idx_del
+        # puts "#{shortvo} #{shortv} \e[34;1m#{action}\e[37;0m #{table} #{args[0]} #{args[1]}"
       end
       this_version_has[action] += 1
     end
@@ -249,6 +258,9 @@ def traverse_all_for_db_schema(app_dir, interval = nil)
       total_action[ac] += num
     end
   end
+  p "#{version_with_change(version_chg)}/#{version_chg.length}"
+  p freq_change_column(column_changes).join "/"
+  p total_action
 end
 
 def text_typed_indexes(app_dir, interval = nil)
